ingester-rpc/src/main/java/io/greptime/rpc/Context.java

```diff
@@ -50,6 +52,28 @@ public static Context of(String key, Object value) {
         return new Context().with(key, value);
     }
 
+    /**
+     * Sets the compression for the context.
+     *
+     * @param compression the compression type
+     * @return this {@link Context}
+     */
+    public Context withCompression(Compression compression) {
+        synchronized (this) {
+            this.compression = compression;
+        }
+        return this;
+    }
+
+    /**
+     * Gets the compression for the context.
+     *
+     * @return the compression type
+     */
+    public Compression getCompression() {
+        return this.compression;
```

Comment:
The `compression` variable must be declared `volatile`; otherwise, the `synchronized` in `withCompression` is meaningless. If `volatile` is used, `synchronized` can be omitted.

Reply:
Good catch, forgot `synchronized`